contrib/internal/httptrace/httptrace.go

```diff
@@ -11,15 +11,35 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"os"
 	"strconv"
 	"strings"
 
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo/instrumentation/httpsec"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var (
+	cfg = newConfig()
+	// envTraceClientIPEnabled is the name of the env var used to specify whether or not to collect client ip in span tags
+	envTraceClientIPEnabled = "DD_TRACE_CLIENT_IP_ENABLED"
 )
 
-var cfg = newConfig()
+// collectClientIP checks if the DD_TRACE_CLIENT_IP_ENABLED environment variable is set to true.
+// returns false when DD_TRACE_CLIENT_IP_ENABLED is unset, set to false, or cannot be parsed to a boolean value
+func collectClientIP() (enabled bool, err error) {
+	if enabledStr, set := os.LookupEnv(envTraceClientIPEnabled); !set {
+		return false, nil
+	} else if enabled, err = strconv.ParseBool(enabledStr); err != nil {
+		return false, fmt.Errorf("could not parse %s value `%s` as a boolean value", envTraceClientIPEnabled, enabledStr)
+	} else {
+		return enabled, nil
+	}
+}
```

Comment:
NIT: This looks a lot like `internal.BoolEnv`, maybe you could (re)use that instead?

https://github.com/DataDog/dd-trace-go/blob/74a8e95a31d9fc95d2f1a7b804fabe7f2467cb0c/internal/env.go#L16-L29